### PR TITLE
[Exports] Correctly type data in excel exports

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAASP-10149

##### SUMMARY
Another go at data typing fields in excel exports. This time instead of relying on other parts of HQ to tell me what data type something is (see https://github.com/dimagi/commcare-hq/pull/26186) I instead employ some regexes and `ValueError` escapes to correctly determine what kind of data is being inserted into a cell, regardless of the question type.

Supports:
- integers
- decimals
- booleans
- dates
- datetimes
- percentage
- comma-separated (US-style) big numbers. (eg `1,000,000.00`)
- dot-separated (EURO-style) big numbers. (eg `1.000.000,00`)
- USD currency (`$1,000.00`)
- EUR currency (`€1.000,00`)

### CODE REVIEW

File: corehq/ex-submodules/couchexport/util.py [hunk 1]
```diff
@@ -175,3 +181,114 @@ def to_json(self, value):
         else:
             function = value
         return function.dumps()
+
+
+def get_excel_format_value(value):
```
Comment (millerdev) [inline corehq/ex-submodules/couchexport/util.py]: Would be great to have some tests to verify that various input values for this function produce the expected output format/value.

**Review by millerdev — APPROVED** [review]

File: corehq/ex-submodules/couchexport/util.py (hunk 1 above)
Comment (biyeun) [reply corehq/ex-submodules/couchexport/util.py]: yes! I think this is a great idea. doing this rn

Comment (biyeun) [reply corehq/ex-submodules/couchexport/util.py]: added some tests @millerdev 😄

File: corehq/ex-submodules/couchexport/tests/test_excel_format_value.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+        excel_format, value = get_excel_format_value('-234')
+        self.assertEqual(excel_format, numbers.FORMAT_NUMBER)
+        self.assertEqual(value, -234)
+        self.assertEqual(type(value), int)
+
+        excel_format, value = get_excel_format_value(324)
+        self.assertEqual(excel_format, numbers.FORMAT_NUMBER)
+        self.assertEqual(value, 324)
+        self.assertEqual(type(value), int)
```
Comment (millerdev) [inline corehq/ex-submodules/couchexport/tests/test_excel_format_value.py]: :heart: Excellent!

nit: these could be DRY'd up quite a bit. One suggestion is to use nose generator test:

```py
from testil import eq

def check(input, output, format, output_type):
    excel_format, value = get_excel_format_value(input)
    eq(excel_format, format)
    eq(value, output)
    eq(type(value), output_type)

def test_integers():
    yield check, '3423', 3423, numbers.FORMAT_NUMBER, int
    yield check, '-234', -234, numbers.FORMAT_NUMBER, int
    yield check, 324, 324, numbers.FORMAT_NUMBER, int

def test_decimal():
    yield check, '4.0345', 4.0345, numbers.FORMAT_NUMBER_00, float
    yield check, '-3.234', -3.234, numbers.FORMAT_NUMBER_00, float
    yield check, '5.032', 5.032, numbers.FORMAT_NUMBER_00, float

...
```

**Review by millerdev — APPROVED** [review]

**Comment by dannyroberts** [comment]
This looks great! One thing I want to make sure we test is the behavior of the "number" column in exports that is used to correlate the separate sheets for repeat questions. Since a question in a repeat group will have any number of responses in a single form, repeat groups are pulled out as separate sheets, and a number column with values like "0.0" and "1.0" are used to correlate the questions to forms which have number column values like "0" and "1". I raise this be cause we don't want "2.0" to be interpreted as a float.

**Comment by biyeun** [comment]
@dannyroberts good point! Right now this would interpret the number column as a float. I'll add that check in

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
🎉 Thanks for writing tests!

**Pull request merged**